alacritty/src/input/keyboard.rs

```diff
@@ -0,0 +1,592 @@
+            // for both small and capital letters on regular characters at the same time.
+            let logical_key = if let Key::Character(ch) = key.logical_key.as_ref() {
+                Key::Character(ch.to_lowercase().into())
+            } else {
+                key.logical_key.clone()
+            };
+
+            let key = match (&binding.trigger, logical_key) {
+                (BindingKey::Scancode(_), _) => BindingKey::Scancode(key.physical_key),
+                (_, code) => BindingKey::Keycode { key: code, location: key.location.into() },
+            };
+
+            if binding.is_triggered_by(mode, mods, &key) {
+                // Pass through the key if any of the bindings has the `ReceiveChar` action.
+                *suppress_chars.get_or_insert(true) &= binding.action != Action::ReceiveChar;
+
+                // Binding was triggered; run the action.
+                binding.action.clone().execute(&mut self.ctx);
+            }
+        }
+
+        suppress_chars.unwrap_or(false)
+    }
+
+    /// Handle key release.
+    fn key_release(&mut self, key: KeyEvent, mode: TermMode, mods: ModifiersState) {
+        if mode.contains(TermMode::REPORT_EVENT_TYPES)
+            && !mode.contains(TermMode::VI)
+            && !self.ctx.search_active()
+            && !self.ctx.display().hint_state.active()
+        {
+            let bytes: Cow<'static, [u8]> = match key.logical_key.as_ref() {
+                // NOTE echo the key back on release to follow kitty/foot behavior. When
+                // KEYBOARD_REPORT_ALL_KEYS_AS_ESC is used, we build proper escapes for
+                // the keys below.
+                _ if mode.contains(TermMode::REPORT_ALL_KEYS_AS_ESC) => {
+                    build_keyboard_esc_sequence(key, mods, mode).into()
+                },
+                // Winit uses different keys for `Backspace` so we expliictly specify the
+                // values, instead of using what was passed to us from it.
+                Key::Named(NamedKey::Tab) => [b'\t'].as_slice().into(),
+                Key::Named(NamedKey::Enter) => [b'\r'].as_slice().into(),
+                Key::Named(NamedKey::Backspace) => [b'\x7f'].as_slice().into(),
+                Key::Named(NamedKey::Escape) => [b'\x1b'].as_slice().into(),
+                _ => build_keyboard_esc_sequence(key, mods, mode).into(),
+            };
+
+            self.ctx.write_to_pty(bytes);
+        }
+    }
+
+    /// Reset search delay.
+    fn reset_search_delay(&mut self) {
+        if self.ctx.search_active() {
+            let timer_id = TimerId::new(Topic::DelayedSearch, self.ctx.window().id());
+            let scheduler = self.ctx.scheduler_mut();
+            if let Some(timer) = scheduler.unschedule(timer_id) {
+                scheduler.schedule(timer.event, TYPING_SEARCH_DELAY, false, timer.id);
+            }
+        }
+    }
+}
+
+/// Build the keyboard escape sequence based on the given `key`, `mods`, and `mode`.
```

Comment:
```suggestion
/// Build a key's escape sequence based on the given `key`, `mods`, and `mode`.
```

It's not an escape sequence for the keyboard, but for a key.